### ISSUE TITLE
Projections are not used in ReplacingMergeTree

### ISSUE THREAD

**Issue by CamiloSierraH** [ClickHouse/ClickHouse#46968]
**Describe the unexpected behaviour**
What is the expected result when defining a Projection in a ReplacingMergeTree table?

Projections could define a different set of keys for the `ORDER BY` and as we are doing a `ReplacingMergeTree` I guess the system when it will merge it will be lost to know which data need to be removed or not, BUT even when setting the same keys (just in a different order) the projections are never used.

If Projections do not work with `ReplacingMergeTree` can we throw an exception when we try to set the projection? (don't get me wrong, if we make it work is even better !! )

**How to reproduce**
* Which ClickHouse server version to use ClickHouse cloud
```
DROP TABLE mySecondReplacingMT;
CREATE TABLE mySecondReplacingMT
(
    `key` Int64,
    `someCol` String,
    `eventTime` DateTime
)
ENGINE = ReplacingMergeTree(eventTime)
ORDER BY (key);

INSERT INTO mySecondReplacingMT Values (1, 'first', '2020-01-01 01:01:01');
INSERT INTO mySecondReplacingMT Values (1, 'second', '2020-01-01 00:00:00');
INSERT INTO mySecondReplacingMT Values (1, 'third', '2020-01-01 02:00:00');
INSERT INTO mySecondReplacingMT Values (2, 'first', '2020-01-01 01:00:00');
INSERT INTO mySecondReplacingMT Values (2, 'second', '2020-01-01 02:00:00');

ALTER TABLE mySecondReplacingMT ADD PROJECTION someCol_proj (
SELECT
* 
ORDER BY (someCol)
);

ALTER TABLE mySecondReplacingMT MATERIALIZE PROJECTION someCol_proj;

EXPLAIN indexes=1 SELECT * FROM mySecondReplacingMT FINAL WHERE someCol='first' ORDER BY someCol;
EXPLAIN indexes=1 SELECT * FROM mySecondReplacingMT FINAL WHERE key=1;
```

**Expected behavior**
Projections work with `ReplacingMergeTree`

**Comment by den-crane** [comment]
* https://github.com/ClickHouse/ClickHouse/issues/24778

**Comment by melvynator** [comment]
Projection should not be used in these situation, we added settings to prevent their usage with some specific table engine: https://github.com/ClickHouse/ClickHouse/pull/66672

**Comment by jihonrado** [comment]
> Projection should not be used in these situation, we added settings to prevent their usage with some specific table engine: [#66672](https://github.com/ClickHouse/ClickHouse/pull/66672)

Shouldn't projections be allowed to be used after #66672 and #62364? Released on [24.08](https://clickhouse.com/blog/clickhouse-release-24-08#control-of-projections-during-merges).

As far as I understood, and following the example in the first post of @CamiloSierraH, deduplication in a `ReplacingMergeTree` will always be done based on the `ORDER BY` specified in the table engine (in this case, `key`) and if something changes in the main table (for instance a new version of the row for `key=second`), the underlying table for the projection `someCol_proj` should be updated accordingly when using `deduplicate_merge_projection_mode=rebuild/drop` (affected part for the row `key=second` is either rebuilt or dropped), to be consistent with the main table.

Am I correct? If this is correct, it would unlock a use case that is pretty common and valuable; consolidating the last version of a mutable event (typically by its id) while allowing a fast access pattern to it based on another field (typically by timestamp). 

Thanks!

**Comment by ssisk** [comment]
I just wanted to second what @jihonrado is saying here - projections are useful/important for handling different searches across large tables, and ReplacingMergeTrees are also important for handling updated data, not being able to use both of them together really limits what we can do.
